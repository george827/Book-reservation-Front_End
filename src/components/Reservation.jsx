import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { cancelReservation } from '../redux/reservations/reservationsSlice';

const Reservation = ({ reservation }) => {
  const dispatch = useDispatch();

  const handleDelete = (id) => {
    dispatch(cancelReservation(id));
    window.location.reload();
  };

  return (
    <div className="my-reservation mt-3 justify-content-between">
      <div className="reservation-image">
        <img src={reservation.restaurant_table.image} alt={reservation.table_name} className="my-reservation-img" />
      </div>
      <div className="ms-3 mt-3 reserve-list">
        <ul className="reserved-items">
          <li>

            <strong>City:</strong>
            <span>{reservation.city}</span>

          </li>

          <li>

            <strong>Start Date:</strong>
            <span>{reservation.start_date}</span>

          </li>

          <li>

            <strong> End Date:</strong>
            <span>{reservation.end_date}</span>

          </li>

          <li>

            <strong>Table Name:</strong>
            <span>{reservation.table_name}</span>

          </li>
          <li>

            <strong>Table size:</strong>
            <span>{reservation.restaurant_table.table_size}</span>

          </li>

        </ul>

        <div className="">
          <button type="button" className="btn btn-danger mt-3" onClick={() => handleDelete(reservation.id)}>Cancel reservation</button>
        </div>
      </div>
    </div>
  );
};

Reservation.propTypes = {
  reservation: PropTypes.shape({
    city: PropTypes.string.isRequired,
    start_date: PropTypes.string.isRequired,
    end_date: PropTypes.string.isRequired,
    table_name: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    restaurant_table: PropTypes.shape({
      image: PropTypes.string.isRequired,
      table_size: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
};

export default Reservation;
