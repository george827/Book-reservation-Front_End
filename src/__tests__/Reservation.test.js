import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Reservation from '../components/Reservation';

const mockStore = configureStore([]);

describe('Reservation component', () => {
  let store;
  let component;

  beforeEach(() => {
    const reservation = {
      id: 1,
      city: 'New York',
      start_date: '2023-06-01',
      end_date: '2023-06-03',
      table_name: 'Table 1',
      restaurant_table: {
        image: 'table1.jpg',
        table_size: 4,
      },
    };

    store = mockStore({
      reservations: {
        reservations: [reservation],
      },
    });

    component = render(
      <Provider store={store}>
        <Reservation reservation={reservation} />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component).toMatchSnapshot();
  });
});
