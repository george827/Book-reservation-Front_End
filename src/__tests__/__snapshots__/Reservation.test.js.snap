// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation component should render with given state from Redux store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-reservation mt-3 justify-content-between"
      >
        <div
          class="reservation-image"
        >
          <img
            alt="Table 1"
            class="my-reservation-img"
            src="table1.jpg"
          />
        </div>
        <div
          class="ms-3 mt-3 reserve-list"
        >
          <ul
            class="reserved-items"
          >
            <li>
              <strong>
                City:
              </strong>
              <span>
                New York
              </span>
            </li>
            <li>
              <strong>
                Start Date:
              </strong>
              <span>
                2023-06-01
              </span>
            </li>
            <li>
              <strong>
                 End Date:
              </strong>
              <span>
                2023-06-03
              </span>
            </li>
            <li>
              <strong>
                Table Name:
              </strong>
              <span>
                Table 1
              </span>
            </li>
            <li>
              <strong>
                Table size:
              </strong>
              <span>
                4
              </span>
            </li>
          </ul>
          <div
            class=""
          >
            <button
              class="btn btn-danger mt-3"
              type="button"
            >
              Cancel reservation
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-reservation mt-3 justify-content-between"
    >
      <div
        class="reservation-image"
      >
        <img
          alt="Table 1"
          class="my-reservation-img"
          src="table1.jpg"
        />
      </div>
      <div
        class="ms-3 mt-3 reserve-list"
      >
        <ul
          class="reserved-items"
        >
          <li>
            <strong>
              City:
            </strong>
            <span>
              New York
            </span>
          </li>
          <li>
            <strong>
              Start Date:
            </strong>
            <span>
              2023-06-01
            </span>
          </li>
          <li>
            <strong>
               End Date:
            </strong>
            <span>
              2023-06-03
            </span>
          </li>
          <li>
            <strong>
              Table Name:
            </strong>
            <span>
              Table 1
            </span>
          </li>
          <li>
            <strong>
              Table size:
            </strong>
            <span>
              4
            </span>
          </li>
        </ul>
        <div
          class=""
        >
          <button
            class="btn btn-danger mt-3"
            type="button"
          >
            Cancel reservation
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
